tests/PHPUnit/Task/Attribute/CreateUpdateDeleteTaskTest.php

```diff
@@ -19,64 +18,48 @@ final class CreateUpdateDeleteTaskTest extends AbstractTaskTest
         $payload = new AttributePayload($this->createClient());
 
+        $setupAttributeTask = $this->taskProvider->get(SetupAttributeTask::class);
+        $setupPayload = $setupAttributeTask->__invoke($payload);
+
         /** @var CreateUpdateEntityTask $task */
         $task = $this->taskProvider->get(CreateUpdateEntityTask::class);
-        $task->__invoke($payload);
-        $finalAttributeCount = self::$container->get('sylius.repository.product_attribute')->count([]);
+        $task->__invoke($setupPayload);
+
+        $finalAttributeCount = $this->getContainer()->get('sylius.repository.product_attribute')->count([]);
 
         $this->assertEquals($attributesCount, $finalAttributeCount);
     }
 
     public function testCreateUpdateTask(): void
     {
         $initialPayload = new AttributePayload($this->createClient());
+        $setupAttributeTask = $this->taskProvider->get(SetupAttributeTask::class);
+        $setupPayload = $setupAttributeTask->__invoke($initialPayload);
+
         /** @var RetrieveAttributesTask $retrieveTask */
         $retrieveTask = $this->taskProvider->get(RetrieveAttributesTask::class);
-        $payload = $retrieveTask->__invoke($initialPayload);
+        $payload = $retrieveTask->__invoke($setupPayload);
 
         /** @var CreateUpdateEntityTask $task */
         $task = $this->taskProvider->get(CreateUpdateEntityTask::class);
         $task->__invoke($payload);
 
+        $tearDownAttributeTask = $this->taskProvider->get(TearDownAttributeTask::class);
+        $tearDownAttributeTask->__invoke($payload);
+
         /** @var \Sylius\Component\Product\Model\ProductAttribute $careInstructionProductAttribute */
-        $careInstructionProductAttribute = self::$container->get('sylius.repository.product_attribute')->findOneBy(['code' => 'care_instructions']);
+        $careInstructionProductAttribute = $this->getContainer()->get('sylius.repository.product_attribute')->findOneBy(['code' => 'care_instructions']);
         $this->assertNotNull($careInstructionProductAttribute);
         $this->assertEquals('Instructions d\'entretien', $careInstructionProductAttribute->getTranslation('fr_FR')->getName());
         $this->assertEquals('Care instructions', $careInstructionProductAttribute->getTranslation('en_US')->getName());
 
         /** @var \Sylius\Component\Product\Model\ProductAttribute $colorProductAttribute */
-        $colorProductAttribute = self::$container->get('sylius.repository.product_attribute')->findOneBy(['code' => 'color']);
+        $colorProductAttribute = $this->getContainer()->get('sylius.repository.product_attribute')->findOneBy(['code' => 'color']);
         $this->assertNotNull($colorProductAttribute);
         $this->assertEquals('Couleur', $colorProductAttribute->getTranslation('fr_FR')->getName());
         $this->assertEquals('Color', $colorProductAttribute->getTranslation('en_US')->getName());
     }
-
-    public function testDeleteTask(): void
```

Comment:
We assume this deletion?

Reply:
yes, it was present but never used. We only do create/update for attributes.